server/index.js

```diff
@@ -2,15 +2,51 @@ require('dotenv/config');
 const express = require('express');
 const staticMiddleware = require('./static-middleware');
 const errorMiddleware = require('./error-middleware');
+const pg = require('pg');
+const ClientError = require('./client-error');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
 
 const app = express();
+app.use(express.json());
 
 app.use(staticMiddleware);
 
 app.get('/api/hello', (req, res) => {
   res.json({ hello: 'world' });
 });
 
+app.post('/api/log/', (req, res, next) => {
+  const sql = `
+  INSERT INTO "logs" ("log", "location", "latitude", "longitude")
+  VALUES             ($1, $2, $3, $4)
+  RETURNING *
+  `;
+
+  const { log, location, latitude, longitude } = req.body;
+  if (!log) {
+    throw new ClientError(400, 'log is required field');
+  }
+  if (!location || !latitude || !longitude) {
+    res.status(500).json({ error: 'an unexpected error occurred' });
+    return;
+  }
```

Comment:
This does not count as `500` (internal server error), instead through a client error with a message about what is wrong with the request body data